src/librustc_mir/transform/const_prop.rs

```diff
@@ -1059,31 +1093,10 @@ impl<'mir, 'tcx> MutVisitor<'tcx> for ConstPropagator<'mir, 'tcx> {
                                 msg,
                             );
                         }
-                    } else {
```

Comment:
if you leave this code in, you get both the mir-opt-level 3 downgrade for the advanced features as well as keep the propagation into asserts and switchint. Leave a FIXME for their removal though